src/main/java/com/hotels/beans/utils/ClassUtils.java

```diff
@@ -142,7 +143,8 @@ public boolean isSpecialType(final Class<?> clazz) {
         notNull(clazz, CLAZZ_CANNOT_BE_NULL);
         final String cacheKey = "isSpecial-" + clazz.getCanonicalName();
         return ofNullable(cacheManager.getFromCache(cacheKey, Boolean.class)).orElseGet(() -> {
-            final Boolean res = clazz.equals(Currency.class) || clazz.equals(Locale.class) || clazz.isSynthetic() || clazz.isAnonymousClass();
+            final Boolean res = clazz.equals(Currency.class) || clazz.equals(Locale.class) || Temporal.class.isAssignableFrom(clazz)
```

Comment:
We could anticipate future additions of more special types by using some sort of `Collection<Type>` to hold them and just check if it contains `clazz`. On second thought, it may be not so straightforward since some checks use equals while others use isAssignable(), so probably we can't use just a `Set.contains()` operation but would have to invoke isAssignable() on every element. This improvement needs to be planned better, ignore it for the moment.

Reply:
The first one is a really good suggestion, we will do it in future releases for sure.
On top of the second one that would use `equals` method rather than `isAssignable()`, I'm afraid this wouldn't work, as it would not work if the class we are checking is an implementation of the interface they are implementing.

**e.g.**
```
Temporal t = Instant.now();
isSpecialType(t.class); // will return true
```
but
```
Instant t = Instant.now();
isSpecialType(t.class); // will return false
```